extensions/amp-video-iframe/1.0/amp-video-iframe.js

```diff
@@ -0,0 +1,93 @@
+/**
+ * Copyright 2020 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import {BUBBLE_MESSAGE_EVENTS} from '../0.1/amp-video-iframe';
+import {VideoBaseElement} from '../../amp-video/1.0/base-element';
+import {VideoIframe} from '../../amp-video/1.0/video-iframe';
+import {createCustomEvent} from '../../../src/event-helper';
+import {dict} from '../../../src/utils/object';
+
+/** @const {string} */
+const TAG = 'amp-video-iframe';
+
+class AmpVideoIframe extends VideoBaseElement {}
+
+/**
+ * @param {!MessageEvent} e
+ */
+const onMessage = (e) => {
+  const method = e.data?.method;
+  if (method) {
+    if (method == 'getIntersection') {
+      // TODO
+      return;
+    }
+    throw new Error(`Unknown method ${method}`);
+  }
+  const event = e.data?.event;
+  if (!event) {
+    return;
+  }
+  if (event === 'analytics') {
+    // TODO
+    return;
+  }
+  if (event === 'error') {
+    // TODO
+    return;
+  }
```

Comment:
This can be handled in one of two ways:

A. Dispatch some sort of custom `error` event.
B. Throw a [`MediaError`](https://developer.mozilla.org/en-US/docs/Web/API/MediaError)

Reply:
What kind of error is this? An error to playback? I'd drive this from whatever HTMLVideoElement is expected to do in this case, which might mean dispatching the `error` event. But the next question is: should the Preact component transition into any particular state due to this error?

Reply:
In classic `amp-video-iframe` this only stops the LOAD signal. I think `error` is fine as a general event, so I added a handler to `VideoWrapper`.